explainability-core/src/main/java/org/kie/trustyai/explainability/local/counterfactual/goal/CounterfactualGoalCriteria.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.trustyai.explainability.local.counterfactual.goal;
+
+import java.util.List;
+
+import org.kie.trustyai.explainability.model.Output;
+
+@FunctionalInterface
+public interface CounterfactualGoalCriteria {
+
+    GoalScore apply(List<Output> prediction);
+
+}
```

Comment:
What about making it extends java Function so that you don't need to declare `apply` again and you also have all the other methods for free (i.e. andThen)?
```suggestion
public interface CounterfactualGoalCriteria extends Function<List<Output>, GoalScore> {
}
```

Reply:
Good idea @danielezonca, this is implemented now.